packages/api/src/utils/site.utils.ts

```diff
@@ -278,6 +278,52 @@ export const hasHoboDataSubQuery = async (
   return hasHoboDataSet;
 };
 
+export const hasWaterQualityDataSubQuery = async (
+  latestDataRepository: Repository<LatestData>,
+): Promise<Map<number, string[]>> => {
+  const latestData: LatestData[] = await latestDataRepository
+    .createQueryBuilder('water_quality_data')
+    .select('site_id', 'siteId')
+    .addSelect('metric')
+    .addSelect('source')
```

Comment:
We can rewrite this to a single `select`

```suggestion
    .select('site_id', 'siteId', 'metric', 'source')
```

Reply:
on a second look, this is needed since the first select also adds alias `site_id` => `sideId`

Reply:
You are right, my bad. Please ignore this comment